app/queries/project_status_query.rb

```diff
@@ -0,0 +1,51 @@
+class ProjectStatusQuery
+  def initialize(platform, requested_project_names)
+    @platform = platform
+    @requested_project_names = requested_project_names
+  end
+
+  def projects_by_name
+    projects.merge(missing_projects)
+  end
+
+  private
+
+  def projects
+    @projects ||= fetch_projects(
+      @requested_project_names,
+      &platform_class.method(:project_find_names)
+    )
+  end
+
+  def missing_projects
+    return {} unless @platform.downcase == "go"
+
+    fetch_projects(
+      @requested_project_names.reject(&projects.method(:key?)),
+      &PackageManager::Go.method(:resolved_name)
+    )
+  end
+
+  def fetch_projects(requested_project_names)
```

Comment:
Should the block be declared in the method signature? I was confused what was going on with the Array yield portion and why it was needed until I saw the calls to this method.

Reply:
The block argument is implicit and idiomatic. It IS possible to specify an argument and `.call` it, but historically that is quite a bit slower (not sure if that has changed in newer Ruby?)

I'd love to make this whole `resolved_project_names` thing easier to read (I've already rebuilt it a few times) - lemme try one more!

Reply:
@mikeyoung85 maybe more readable now?

Reply:
Seems better to me. That slight separation makes it easier to spot the block.